src/pages/Main.tsx

```diff
@@ -1,50 +1,63 @@
 import styled from '@emotion/styled'
 import { useState, useEffect } from 'react'
+import { useQuery } from '@tanstack/react-query'
 import Timer from '../components/Timer'
 import ExerciseList, { Exercise } from '../components/ExerciseList'
-import mainMock from '../mocks/MainMock'
 import DiaryCreate from '../components/DiaryCreate'
 import axiosInstance from '../api/axiosInstance'
 import TodayDiary from '../components/TodayDiary'
+import Error from '../components/Error'
+import Loading from '../components/Loading'
+
+const fetchExercise = async () => {
+  const accessToken = localStorage.getItem('authToken')
+
+  const response = await axiosInstance.get('/api', {
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+    },
+  })
+  return response.data
+}
 
 const Main = () => {
-  const [totalTime, setTotalTime] = useState(mainMock.totalTime)
-  const [exerciseList, setExerciseList] = useState<Exercise[]>(
-    mainMock.exerciseList
-  )
-  const [diary, setDiary] = useState(mainMock.diary)
+  const { data, isLoading, isError } = useQuery({
+    queryKey: ['main'],
+    queryFn: fetchExercise,
+    retry: 1,
+  })
+
+  const [totalTime, setTotalTime] = useState(0)
+  const [exerciseList, setExerciseList] = useState<Exercise[]>([])
+  const [diary, setDiary] = useState([])
```

Comment:
diary가 never로 추론되고 있어서 타입 명시해주는게 좋겠습니다.


```suggestion
  const [diary, setDiary] = useState<Diary[]>([])
```